.github/workflows/_snap-release.yaml

```diff
@@ -0,0 +1,79 @@
+on:
+  workflow_call:
+    inputs:
+      track:
+        description: |
+          The track to which charm should be released. This input supports a
+          comma-separated tracks in string format. The channels will be a list of
+          track/risk-level combinations. By default, the snap will be released to
+          channel(s) in the "latest" track.
+          Examples:
+          - "latest"
+          - "latest,test,2.9"
+        required: false
+        type: string
+        default: latest
+      risk-level:
+        description: |
+          The risk level to which charm should be released. Valid inputs are one or
+          a combination (in comma-separated string format) of: "edge", "beta",
+          "candidate", "stable". The channels will be a list of track/risk-level
+          combinations. By default, the snap will be released to channel(s) with
+          "edge" risk level.
+          Examples:
+          - "edge"
+          - "edge,candidate"
+        required: false
+        type: string
+        default: edge
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: Publish Snap
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: snap-artifact
+      - id: get-snap-name
+        run: echo "SNAP_FILE=$(ls *.snap)" >> $GITHUB_OUTPUT
```

Comment:
Actually snap-artifact is the name of an archive file containing the snap. And when downloading the artifact, the content of it will be decompressed and saved to the local FS. Therefore, it is necessary to get the name of the snap with `ls`